bindings/node/native/src/lib.rs

```diff
@@ -22,102 +21,68 @@ use tokio::runtime::Runtime;
 
 /// Gets the account instances map.
-fn instances() -> &'static AccountInstanceMap {
-    static INSTANCES: Lazy<AccountInstanceMap> = Lazy::new(|| {
-        iota_wallet::event::on_balance_change(|event| {
-            let address = event.cloned_address();
-            let balance = *event.balance();
-            mutate_account_if_exists(event.account_id(), move |account| {
-                let addresses = account.addresses_mut();
-                if let Some(address) = addresses.iter_mut().find(|a| a == &&address) {
-                    address.set_balance(balance);
-                }
-            });
-        });
-        iota_wallet::event::on_new_transaction(|event| {
-            let message = event.cloned_message();
-            mutate_account_if_exists(event.account_id(), move |account| {
-                account.append_messages(vec![message]);
-            });
-        });
-        iota_wallet::event::on_confirmation_state_change(|event| {
-            let message = event.cloned_message();
-            let confirmed = *event.confirmed();
-            mutate_account_if_exists(event.account_id(), move |account| {
-                if let Some(message) = account.messages_mut().iter_mut().find(|m| m == &&message) {
-                    message.set_confirmed(Some(confirmed));
-                }
-            });
-        });
-        iota_wallet::event::on_reattachment(|event| {
-            let message = event.cloned_message();
-            mutate_account_if_exists(event.account_id(), move |account| {
-                account.append_messages(vec![message]);
-            });
-        });
-        iota_wallet::event::on_broadcast(|event| {
-            let message = event.cloned_message();
-            mutate_account_if_exists(event.account_id(), move |account| {
-                if let Some(message) = account.messages_mut().iter_mut().find(|m| m == &&message) {
-                    message.set_broadcasted(true);
-                }
-            });
-        });
-        Default::default()
-    });
+fn account_instances() -> &'static AccountInstanceMap {
+    static INSTANCES: Lazy<AccountInstanceMap> = Lazy::new(Default::default);
     &INSTANCES
 }
 
-pub(crate) fn get_account(id: &str) -> Arc<RwLock<Account>> {
-    let map = instances()
+pub(crate) fn get_account(id: &AccountIdentifier) -> AccountGuard {
+    let map = account_instances()
         .read()
         .expect("failed to lock account instances: get_account()");
     map.get(id).expect("account dropped or not initialised").clone()
 }
 
-pub(crate) fn store_account(account: Account) -> String {
-    let mut map = instances()
+pub(crate) fn store_account(account: AccountGuard) -> AccountIdentifier {
+    let mut map = account_instances()
         .write()
         .expect("failed to lock account instances: store_account()");
-    let id: String = thread_rng().sample_iter(&Alphanumeric).take(10).collect();
-    map.insert(id.clone(), Arc::new(RwLock::new(account)));
+    let id = {
```

Comment:
I'm curious: can this not be done inline?

Reply:
oh you're right, I guess there's a lot of places I can use it. I thought I needed to save the reference so it doesn't go out of scope.